vcd/metadata.go

```diff
@@ -92,18 +214,9 @@ func metadataEntryResourceSchema(objectNameInDescription string) *schema.Schema
 	}
 }
 
-// getMetadataEntrySchema returns a schema for the "metadata_entry" attribute, that can be used to
-// build data sources (isDatasource=true) or resources (isDatasource=false). The description of the
-// attribute will refer to the input resource name.
-func getMetadataEntrySchema(resourceNameInDescription string, isDatasource bool) *schema.Schema {
```

Comment:
Just curious, what prompted you to remove this function and use the two specific ones below across the board? :)

Reply:
When I implemented that I thought it was a good idea, but after revisiting this function to implement this feature I noticed:
- The bool parameter is hard to understand unless you read and understand the function docs/signature.
- The bool parameter is a bit arbitrary, it could have been `isResource`, which also makes it confusing.
- The function just had an `if` based on that condition that chose either the resource schema or datasource schema.

So it seemed more right to me to have just more simple and meaningful functions `metadataEntryDatasourceSchema` and `metadataEntryResourceSchema`. This way you explicitly know what you are trying to get, without an opaque bool flag.

Reply:
As a side note, I also thought of using the same pattern we use across the codebase, something like `getMetadataEntrySchema("foo", "resource") and getMetadataEntrySchema("foo", "datasource")`, but again the inner body was so dumb that it didn't have much sense to have a 2x1 function.